### PR TITLE
Update upstream

### PR DESCRIPTION
**Pull request by hervenicol**
Update to current upstream helm chart.

Towards https://github.com/giantswarm/giantswarm/issues/23494

### CODE REVIEW

**Comment by hervenicol** [comment]
ok, let me fix the build before reviewing :facepalm:

**Review by QuentinBisson — APPROVED** [review]

**Comment by hervenicol** [comment]
Ugh, test-deploying it on Gauss fails on servicemonitors / kyverno rules :crying_cat_face: :
```
$ opsctl deploy --use-kubeconfig grafana@update-upstream
Requesting the creation of deployment using App method 

Suspended collection and flux kustomizations

    chart           grafana
    repository      grafana-app
    ref             update-upstream
    version         1.0.3-03957eee0b9a9bd0a9ec42e3e79cf83719e4d965
    installation    

Requested the creation of deployment using App method

Waiting for deployment 
Failed.
Error: Execution failed: deployment failed (status: `failed`, reason: `Upgrade "grafana" failed: cannot patch "grafana" with kind ServiceMonitor: Internal error occurred: error in add for path: '/spec/endpoints/0/relabelings/20': Unable to access invalid index: 20: invalid index referenced`)
        /home/runner/work/opsctl/opsctl/pkg/cmd/deploy/deploy.go:389
        /home/runner/work/opsctl/opsctl/pkg/cmd/deploy/appdeploy/deployer.go:284
        /home/runner/go/pkg/mod/github.com/giantswarm/backoff@v1.0.0/retry.go:13
        /home/runner/work/opsctl/opsctl/pkg/cmd/deploy/appdeploy/deployer.go:293
        /home/runner/work/opsctl/opsctl/command/deploy/command.go:364
        /home/runner/work/opsctl/opsctl/pkg/output/funcs.go:33
        /home/runner/work/opsctl/opsctl/command/deploy/command.go:374
        /home/runner/work/opsctl/opsctl/command/deploy/command.go:146
        /home/runner/work/opsctl/opsctl/main.go:57
```

File: helm/grafana/templates/image-renderer-network-policy.yaml [hunk 1]
```diff
@@ -64,10 +64,7 @@ spec:
         - port: {{ .Values.service.port }}
           protocol: TCP
       to:
-        - namespaceSelector:
-            matchLabels:
-              name: {{ template "grafana.namespace" . }}
```
Comment (TheoBrigitte) [inline helm/grafana/templates/image-renderer-network-policy.yaml:69]: Feels strange that the namespaceSelector is gone here but not on other NetworkPolicies

**Review by TheoBrigitte — APPROVED** [review]

**Comment by hervenicol** [comment]
Deleted the `grafana` servicemonitor, redeployed: failed again.
But retried deploying, and it eventually worked :thinking: :question: :question: :exclamation:

File: helm/grafana/templates/image-renderer-network-policy.yaml (hunk 1 above)
Comment (hervenicol) [reply helm/grafana/templates/image-renderer-network-policy.yaml:69]: I checked, and that's a legit upstream change: issue https://github.com/grafana/helm-charts/issues/795 fixed by https://github.com/clementnuss/helm-charts/commit/82b0677d08a58977719fe5ce20e0ba140d51c3fc

**Comment by mcbenjemaa** [comment]
> Deleted the `grafana` servicemonitor, redeployed: failed again. But retried deploying, and it eventually worked 🤔 ❓ ❓ ❗

Have you tried to delete the app from scratch?
Maybe some caches!

**Pull request merged**